velox/exec/fuzzer/WriterFuzzer.cpp

```diff
@@ -328,21 +334,36 @@ void WriterFuzzer::verifyWriter(
       assertEqualResults(expectedResult, plan->outputType(), {result}),
       "Velox and reference DB results don't match");
 
-  // 2. Verify directory layout.
+  // 2. Verifies directory layout.
   const auto referencedOutputDirectoryPath =
       getReferenceOutputDirectoryPath(partitionKeys.size());
   comparePartitions(outputDirectoryPath, referencedOutputDirectoryPath);
 
+  // 3. Verifies data itself.
+  auto splits = makeSplits(input, outputDirectoryPath, rootPool_);
+  auto readPlan =
+      PlanBuilder().tableScan(asRowType(input[0]->type())).planNode();
+  auto actual = execute(readPlan, maxDrivers, splits);
+  auto reference_data =
+      referenceQueryRunner_->execute("SELECT * FROM tmp_write");
+  VELOX_CHECK(
+      assertEqualResults(reference_data, {actual}),
+      "Velox and reference DB results don't match");
+
   LOG(INFO) << "Verified results against reference DB";
 }
 
 RowVectorPtr WriterFuzzer::execute(
     const core::PlanNodePtr& plan,
-    int32_t maxDrivers) {
+    const int32_t maxDrivers,
+    const std::vector<exec::Split>& splits) {
   LOG(INFO) << "Executing query plan: " << std::endl
             << plan->toString(true, true);
   fuzzer::ResultOrError resultOrError;
   AssertQueryBuilder builder(plan);
+  if (!splits.empty()) {
```

Comment:
in which case splits is empty?

Reply:
When we execute a plan with Value Node instead of Table Scan Node